source/portal/socketIOServer.js

```diff
@@ -352,14 +352,22 @@ var SocketIOServer = function(spec, portal, observer) {
     sioOptions.pingTimeout = spec.pingTimeout * 1000;
   }
 
-  var startInsecure = function(port) {
+  var startInsecure = function(port, cors) {
     var server = require('http').createServer().listen(port);
     io = require('socket.io').listen(server, sioOptions);
+    io.origins((origin, callback) => {
+      if(cors !== '*'){
```

Comment:
Use RegExp to match origin here instead of '!==' check. Strings like "127.0.0.*" should be supported.
You can refer to 'replaced_ip_address' configuration in webrtc_agent/agent.toml.